### PR TITLE
[W6.4d][F12-B4]Zhang Ruiqi

### PR DESCRIPTION
**Pull request by zhangriqi**
Add polymorphism for Command abstract class and its subclasses

### CODE REVIEW

**Comment by zhangriqi** [comment]
When I run the Main method in IntelliJ, I gets "Exception in Application start method". Any one has the same problem?

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#Fri Feb 23 09:23:35 CST 2018
+distributionBase=GRADLE_USER_HOME
```
Comment (OuyangDanwen) [inline gradle/wrapper/gradle-wrapper.properties:2]: Why this file appears?

File: gradlew [hunk 2]
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env sh
```
Comment (OuyangDanwen) [inline gradlew:1]: Same here.

File: test/java/seedu/addressbook/logic/LogicTest.java [hunk 3]
```diff
@@ -86,11 +87,14 @@ private void assertCommandBehavior(String inputCommand,
...
             assertEquals(lastShownList, r.getRelevantPersons().get());
         }
-
+        //Parse the user command
+        Command command = new Parser().parseCommand(inputCommand);
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (command.isMutating()) {
```
Comment (OuyangDanwen) [inline test/java/seedu/addressbook/logic/LogicTest.java:95]: How about the else branch? How would you go about the case where the command is not mutating?

File: src/seedu/addressbook/commands/Command.java [hunk 4]
```diff
@@ -70,4 +70,12 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    /**
+     *
+     * @return whether a command is mutating the data in the addressbook
```
Comment (OuyangDanwen) [inline src/seedu/addressbook/commands/Command.java:76]: Try to write the header comment like this: returns true if the command ... ...

**Review by OuyangDanwen — COMMENTED** [review]
Generally okay. Testing is incomplete. Try to improve it. Also, do not include redundant files in you PR. Check also what caused your build to fail with travis and try to fix it. You can close the PR after reading the comments. @zhangriqi

**Pull request closed without merging**